WalletWasabi.Gui/Controls/WalletExplorer/CoinListViewModel.cs

```diff
@@ -197,37 +177,21 @@ public GridLength CoinJoinStatusWidth
 		private bool? GetCheckBoxesSelectedState(Func<CoinViewModel, bool> coinFilterPredicate)
 		{
 			var coins = Coins.Where(coinFilterPredicate).ToArray();
-			bool IsAllSelected = true;
-			foreach (CoinViewModel coin in coins)
-			{
-				if (!coin.IsSelected)
-				{
-					IsAllSelected = false;
-					break;
-				}
-			}
+			bool isAllSelected = coins.All(coin => coin.IsSelected);
 
-			bool IsAllDeselected = true;
-			foreach (CoinViewModel coin in coins)
-			{
-				if (coin.IsSelected)
-				{
-					IsAllDeselected = false;
-					break;
-				}
-			}
+			bool isAllDeselected = coins.All(coin => !coin.IsSelected);
 
-			if (IsAllDeselected)
+			if (isAllDeselected)
 			{
 				return false;
 			}
 
-			if (IsAllSelected)
+			if (isAllSelected)
 			{
 				return true;
 			}
 
-			return null;
+			return false;
```

Comment:
this could be replaced by `return isAllSelected && !isAllDeselected`

Reply:
Done.